src/components/Pagination/Pagination.tsx

```diff
@@ -32,145 +32,106 @@ export type PaginationProps = {
-      ariaLabel: 'Previous',
-      additionalClassName: this.props.cssClasses.previousPageItem,
-      isDisabled: this.props.isFirstPage,
-      label: this.props.templates.previous,
-      pageNumber: this.props.currentPage - 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private nextPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'Next',
-      additionalClassName: this.props.cssClasses.nextPageItem,
-      isDisabled: this.props.isLastPage,
-      label: this.props.templates.next,
-      pageNumber: this.props.currentPage + 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private firstPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'First',
-      additionalClassName: this.props.cssClasses.firstPageItem,
-      isDisabled: this.props.isFirstPage,
-      label: this.props.templates.first,
-      pageNumber: 0,
-      createURL: this.props.createURL,
-    });
+    props.setCurrentPage(pageNumber);
   };
 
-  private lastPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'Last',
-      additionalClassName: this.props.cssClasses.lastPageItem,
-      isDisabled: this.props.isLastPage,
-      label: this.props.templates.last,
-      pageNumber: this.props.nbPages! - 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private pages = () => {
-    return this.props.pages!.map((pageNumber) =>
-      this.pageLink({
-        ariaLabel: `${pageNumber + 1}`,
-        additionalClassName: this.props.cssClasses.pageItem,
-        isSelected: pageNumber === this.props.currentPage,
-        label: `${pageNumber + 1}`,
-        pageNumber,
-        createURL: this.props.createURL,
-      })
-    );
-  };
-
-  public render() {
-    return (
-      <div
-        className={cx(this.props.cssClasses.root, {
-          [this.props.cssClasses.noRefinementRoot]: this.props.nbPages! <= 1,
-        })}
-      >
-        <ul className={this.props.cssClasses.list}>
-          {this.props.showFirst && this.firstPageLink()}
-          {this.props.showPrevious && this.previousPageLink()}
-          {this.pages()}
-          {this.props.showNext && this.nextPageLink()}
-          {this.props.showLast && this.lastPageLink()}
-        </ul>
-      </div>
-    );
-  }
+  const pages = props.pages!.map((pageNumber) => (
+    <PaginationLink
+      key={pageNumber}
+      ariaLabel={`${pageNumber + 1}`}
+      additionalClassName={props.cssClasses.pageItem}
+      isSelected={pageNumber === props.currentPage}
+      label={`${pageNumber + 1}`}
+      pageNumber={pageNumber}
+      createURL={props.createURL}
+      cssClasses={props.cssClasses}
+      handleClick={handleClick}
+    />
+  ));
+
+  return (
+    <div
+      className={cx(props.cssClasses.root, {
+        [props.cssClasses.noRefinementRoot]: props.nbPages! <= 1,
+      })}
+    >
+      <ul className={props.cssClasses.list}>
+        {props.showFirst && (
+          <PaginationLink
+            key="first"
```

Comment:
Why do we need a key here since we're not in a list? Same question for previous, next and last.

Reply:
we actually are in a list: `[first, previous, ...pages, next, last]`

Reply:
This element is not part of an array, is it? It should be stable.

Reply:
It is not needed here, as @sarahdayan explained.